synapse/handlers/room.py

```diff
@@ -1768,6 +1807,63 @@ class ShutdownRoomResponse(TypedDict):
     new_room_id: Optional[str]
 
 
+@attr.s(slots=True, auto_attribs=True)
+class DeleteStatus:
+    """Object tracking the status of a delete room request
+
+    This class contains information on the progress of a delete room request, for
+    return by get_delete_status.
+    """
+
+    ACTION_SHUTDOWN = "shutdown"
```

Comment:
We could make these more obvious:

```suggestion
    ACTION_SHUTDOWN_AND_PURGE_ROOM = "shutdown_and_purge_room"
```